Admin/Test-AMSI.ps1

```diff
@@ -1,144 +1,782 @@
+                }
+            } else {
+                $remoteExchangePath = (Invoke-Command -ComputerName $ExchangeServer -ScriptBlock { (Get-ChildItem Env:ExchangeInstallPath).Value }  -ArgumentList $env:ExchangeInstallPath -ErrorAction SilentlyContinue -ErrorVariable InvokeError)
+                if ( $remoteExchangePath ) {
+                    $FEEcpWebConfig = Join-Path "\\$ExchangeServer\$($remoteExchangePath.Replace(':','$'))" "FrontEnd\HttpProxy\ecp\web.config"
+                    $CAEEcpWebConfig = Join-Path "\\$ExchangeServer\$($remoteExchangePath.Replace(':','$'))" "ClientAccess\ecp\web.config"
+                }
+            }
+
+            if ( $FEEcpWebConfig -and $CAEEcpWebConfig) {
+                if ( Test-Path $FEEcpWebConfig -PathType Leaf) {
+                    $FEFilterModule = $null
+                    $FEFilterModule = Get-Content $FEEcpWebConfig | Where-Object { $_ -match '<add name="HttpRequestFilteringModule" type="Microsoft.Exchange.HttpRequestFiltering.HttpRequestFilteringModule, Microsoft.Exchange.HttpRequestFiltering, Version=15.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35"' }
+                    Write-Output ""
+                    if ( $FEFilterModule ) {
+                        Write-Host "We found HttpRequestFilteringModule on FrontEnd ECP web.config" -ForegroundColor Green
+                        Write-Host "on: %ExchangeInstallPath%FrontEnd\HttpProxy\ecp\web.config"
+                    } else {
+                        Write-Warning "We did not find HttpRequestFilteringModule on FrontEnd ECP web.config"
+                        Write-Warning "on: %ExchangeInstallPath%FrontEnd\HttpProxy\ecp\web.config"
+                    }
+                } else {
+                    Write-Warning "We did not find web.config for FrontEnd ECP"
+                    Write-Warning "on: %ExchangeInstallPath%FrontEnd\HttpProxy\ecp\web.config"
+                }
+
+                if ( Test-Path $FEEcpWebConfig -PathType Leaf) {
+                    $CEFilterModule = $null
+                    $CEFilterModule = Get-Content $CAEEcpWebConfig | Where-Object { $_ -match '<add name="HttpRequestFilteringModule" type="Microsoft.Exchange.HttpRequestFiltering.HttpRequestFilteringModule, Microsoft.Exchange.HttpRequestFiltering, Version=15.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35"' }
+                    Write-Output ""
+                    if ( $CEFilterModule ) {
+                        Write-Host "We found HttpRequestFilteringModule on ClientAccess ECP web.config" -ForegroundColor Green
+                        Write-Host "on %ExchangeInstallPath%ClientAccess\ecp\web.config"
+                    } else {
+                        Write-Warning "We did not find HttpRequestFilteringModule on ClientAccess ECP web.config"
+                        Write-Warning "on %ExchangeInstallPath%ClientAccess\ecp\web.config"
```

Comment:
Use the actual location stored in variable.

Reply:
In previous PR I used the variable, but I was not sure it is better and finally I change it. 
I will change back.